kube-runtime/src/watcher.rs

```diff
@@ -264,3 +273,32 @@ pub fn watch_object<K: Resource + Clone + DeserializeOwned + Debug + Send + 'sta
         Event::Applied(obj) => Ok(Some(obj)),
     })
 }
+
+/// Pauses the stream for a while (as defined by `backoff`) after each [`Err`].
+pub fn backoff_watch<K, B>(
+    stream: impl Stream<Item = Result<Event<K>>> + Send,
+    backoff: B,
+) -> impl Stream<Item = Result<Event<K>>> + Send
+where
+    K: Resource + Send,
+    B: Backoff + Send,
+{
+    StreamBackoff::new(stream, backoff)
+}
```

Comment:
This appears unused now after the `Observer` cancel. I can remove this?

Reply:
Good catch. This was an attempt to be somewhat more ergonomic and point people who were already working in the `watcher` module towards it (rather than focusing on the implementation details of `StreamBackoff` itself). We can either drop this, or revert `StreamBackoff::new` to `pub(crate)` and fixing the examples to use `backoff_watch` instead.

I'm fine with either.

Reply:
I removed it for now. Want to focus on the chain pipeline before teaching people too much new stuff.